src/main/java/com/exasol/logging/VersionCollector.java

```diff
@@ -0,0 +1,35 @@
+package com.exasol.logging;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class VersionCollector {
+    private static final String PATH = "META-INF/maven/com.exasol/virtual-schema-common-java/pom.properties";
+    private static final String VERSION = "version";
+    private final String path;
+
+    public VersionCollector(final String path) {
+        this.path = path;
+    }
+
+    public VersionCollector() {
+        this.path = PATH;
+    }
+
+    public String getVersionNumber() {
+        final Properties properties = new Properties();
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+        final InputStream stream = loader.getResourceAsStream(this.path);
+        if (stream == null) {
+            return "Unable to read the version. Check the file path.";
```

Comment:
If the method is called `getVersionNumber()` you can either:

* Return the actual version number
* Throw and exception
* Return a placeholder like 'UNKNOWN'

Returning a sentence instead breaks the SRP. I think it is better to throw a derived runtime exception if the version number can't be read because that means the package is not built correctly (a programming error if you will).